app/controllers/names/descriptions_controller.rb

```diff
@@ -79,11 +79,10 @@ def list_name_descriptions
 
     # Display list of name_descriptions that a given user is author on.
     def name_descriptions_by_author
-      user = if params[:by_author]
-               find_or_goto_index(User, params[:by_author].to_s)
-             else
-               @user
-             end
+      user = find_obj_or_goto_index(
+        model: User, obj_id: params[:by_author].to_s,
+        index_path: name_descriptions_path
+      )
```

Comment:
Changes functionality when by_user is not an existing user.
See [Comment about LocationsController](https://github.com/MushroomObserver/mushroom-observer/pull/1324/files#r1095161355).